modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-fragment.glsl.js

```diff
@@ -36,23 +37,30 @@ varying vec2 uv;
 void main(void) {
   geometry.uv = uv;
 
-  vec4 texColor = texture2D(iconsTexture, vTextureCoords);
-  
-  float alpha = texColor.a;
+  float alpha = texture2D(iconsTexture, vTextureCoords).a;
+
   // if enable sdf (signed distance fields)
   if (sdf) {
-    float distance = texture2D(iconsTexture, vTextureCoords).a;
-    alpha = smoothstep(buffer - vGamma, buffer + vGamma, distance);
+    alpha = smoothstep(buffer - vGamma, buffer + vGamma, alpha);
   }
 
   // Take the global opacity and the alpha from vColor into account for the alpha component
   float a = alpha * vColor.a;
-
+  
   if (a < alphaCutoff) {
-    discard;
+    if ((backgroundColor.a < 0.001) || picking_uActive) {
+      discard;
+    } else {
+      gl_FragColor = vec4(backgroundColor.rgb, backgroundColor.a * vColor.a);
+      return;
+    }
   }
 
-  gl_FragColor = vec4(vColor.rgb, a);
+  if (backgroundColor.a < 0.001) {
```

Comment:
Make a variable `bool shouldDrawBackground`

Reply:
uniform, set from the layer code?

Reply:
Done as uniform